File: Array/3375. Minimum Operations to Make Array Values Equal to K.cpp
/*
3375. Minimum Operations to Make Array Values Equal to K
*/

// TC:O(n) SC:O(n)
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        // sort(nums.rbegin(), nums.rend());
        // if(nums.back() < k) return -1;
        int n = nums.size();
        int res = 0;
        int minVal = INT_MAX;
        unordered_set<int> set;
        for(int i = 0; i < n; i++) {
            set.insert(nums[i]);
            minVal = min(minVal, nums[i]);
        }
        if(minVal < k) return -1;
        res += (int)set.size() - (minVal == k ? 1 : 0);
        
        return res;
    }
};

File: Array/3375. Minimum Operations to Make Array Values Equal to K.py
# TC:O(n) SC:O(U) U為數字種類
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        '''
        就是看一開始有多少數字比k大 如果有比k小的就無法達成
        '''
        if min(nums) < k:
            return -1
        return len(set(nums)) + (min(nums) > k) - 1
